tests/CMakeLists.txt

```diff
@@ -108,7 +125,11 @@ pybind11_add_module(pybind11_tests pybind11_tests.cpp
 pybind11_enable_warnings(pybind11_tests)
 
 if(EIGEN3_FOUND)
-  target_include_directories(pybind11_tests PRIVATE ${EIGEN3_INCLUDE_DIR})
+  if (PYBIND11_EIGEN_VIA_TARGET)
+    target_link_libraries(pybind11_tests PRIVATE Eigen3::Eigen)
```

Comment:
Okay; I'll move to CMAKE version guard around the CONFIG loader.

I like the idea of keeping the updated CONFIG loader and cmake target requirements here--upstream eigen has declared adding the include path manually to be deprecated.  It's unlikely to actually matter in practice, of course, but it seems a bit cleaner to not use a deprecated interface if we can easily avoid it.